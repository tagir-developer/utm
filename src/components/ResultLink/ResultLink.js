import React from 'react'

const ResultLink = ({ text, title, resultLink, resultLinkChange, copyResultLink }) => {

	const id = 'result-link-input'

	const resultLinkHandler = () => {
		return copyResultLink(id)
	}

	return (
		<div className="mt-3">
			{/* <label htmlFor={`basic-url-${Math.random()}`} className="font-weight-bold">{title}</label> */}
			<label htmlFor="result-link-input" className="font-weight-bold">{title}</label>
			<div className="input-group">
				<div className="input-group-prepend">
					<button
						className="btn btn-primary"
						type="button"
						id="result-link-button"
						onClick={resultLinkHandler}
					>
						Скопировать
					</button>
				</div>
				<input
					type="text"
					id="result-link-input"
					className="form-control"
					placeholder={text}
					aria-label="Example text with button addon"
					aria-describedby="result-link-button"
					value={resultLink}
					onChange={e => resultLinkChange(e)}
				/>
			</div>
		</div>
	)
}

ResultLink.defaultProps = {
	text: 'Значение плейсхолдера', 
	title: 'Дефолтный заголовок', 
	resultLink: 'Итоговая ссылка', 
	resultLinkChange: () => {}, 
	copyResultLink: () => {},
}

export default ResultLink