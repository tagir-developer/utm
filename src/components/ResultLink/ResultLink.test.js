import React from 'react'
import ResultLink from './ResultLink'


describe('<ResultLink />', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<ResultLink />)
	})

	describe('Component render. Component props testing', () => {

		it('Should ResultLink component render with default props', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})

		it('Should ResultLink component render with test props', () => {
			wrapper.setProps({
				text: 'Тестовый плейсхолдер',
				title: 'Тестовый заголовок',
				resultLink: 'Тестовая ссылка',
				resultLinkChange: () => {},
				copyResultLink: () => {},
			})
			expect(toJson(wrapper)).toMatchSnapshot()
		})

	})

	describe('ResultLink component handlers testing', () => {

		it('Should call "resultLinkChange"', () => {
			const changeHandler = jest.fn()
			wrapper.setProps({
				resultLinkChange: changeHandler,
			})
			expect(changeHandler).toHaveBeenCalledTimes(0)
			wrapper.find('input').simulate('change')
			expect(changeHandler).toHaveBeenCalled()
		})

		it('Should call "copyResultLink"', () => {
			const changeHandler = jest.fn()
			wrapper.setProps({
				copyResultLink: changeHandler,
			})
			expect(changeHandler).toHaveBeenCalledTimes(0)
			wrapper.find('button').simulate('click')
			expect(changeHandler).toHaveBeenCalled()
		})

		it('Should use default handler "resultLinkChange"', () => {
			const result = ResultLink.defaultProps.resultLinkChange()
			expect(result).toBe(undefined)
		})

		it('Should use default handler "copyResultLink"', () => {
			const result = ResultLink.defaultProps.copyResultLink()
			expect(result).toBe(undefined)
		})

	})


})