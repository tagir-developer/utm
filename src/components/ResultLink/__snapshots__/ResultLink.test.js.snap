// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultLink /> Component render. Component props testing Should ResultLink component render with default props 1`] = `
<div
  className="mt-3"
>
  <label
    className="font-weight-bold"
    htmlFor="result-link-input"
  >
    Дефолтный заголовок
  </label>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <button
        className="btn btn-primary"
        id="result-link-button"
        onClick={[Function]}
        type="button"
      >
        Скопировать
      </button>
    </div>
    <input
      aria-describedby="result-link-button"
      aria-label="Example text with button addon"
      className="form-control"
      id="result-link-input"
      onChange={[Function]}
      placeholder="Значение плейсхолдера"
      type="text"
      value="Итоговая ссылка"
    />
  </div>
</div>
`;

exports[`<ResultLink /> Component render. Component props testing Should ResultLink component render with test props 1`] = `
<div
  className="mt-3"
>
  <label
    className="font-weight-bold"
    htmlFor="result-link-input"
  >
    Тестовый заголовок
  </label>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <button
        className="btn btn-primary"
        id="result-link-button"
        onClick={[Function]}
        type="button"
      >
        Скопировать
      </button>
    </div>
    <input
      aria-describedby="result-link-button"
      aria-label="Example text with button addon"
      className="form-control"
      id="result-link-input"
      onChange={[Function]}
      placeholder="Тестовый плейсхолдер"
      type="text"
      value="Тестовая ссылка"
    />
  </div>
</div>
`;
